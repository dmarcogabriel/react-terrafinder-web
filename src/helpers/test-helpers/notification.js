export const mockUseNotification = (showNotification = () => {}) => {
  const mockShowNotification = jest.fn(showNotification);

  jest.mock('hooks/useNotification', () => ({
    useNotification: () => ({
      showNotification: mockShowNotification,
    }),
    NOTIFICATION_TYPES: { SUCCESS: 'success', ERROR: 'error' },
  }));

  return { mockShowNotification };
};
