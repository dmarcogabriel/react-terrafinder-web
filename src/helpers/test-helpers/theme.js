import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/themes';

export const withTheme = (component) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);

export const renderWithTheme = (component) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
