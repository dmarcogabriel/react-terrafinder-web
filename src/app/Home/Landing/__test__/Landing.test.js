import React from 'react';
import ReactDOM from 'react-dom';
import { renderWithTheme, withTheme } from 'helpers/test-helpers/theme';
// import { Landing } from '..';

describe('<Landing />', () => {
  it('renders without crashing', () => {
    // const div = document.createElement('div');
    // ReactDOM.render(withTheme(<Landing />), div);
    // ReactDOM.unmountComponentAtNode(div);
  });

  // todo: formik problems
  // it('should handle filters event', () => {
  //   const mockFn = jest.fn();
  //   jest.doMock('react-router-dom', () => ({
  //     useHistory: () => ({
  //       push: mockFn,
  //     }),
  //   }));

  //   const { getByTestId } = render(<Landing />);

  //   const filterButton = getByTestId('submitButton');
  //   fireEvent.click(filterButton);

  //   expect(mockFn).toHaveBeenCalled();
  // });

  it('matches snapshot', () => {
    // const tree = renderWithTheme(<Landing />).toJSON();
    // expect(tree).toMatchSnapshot();
  });
});
