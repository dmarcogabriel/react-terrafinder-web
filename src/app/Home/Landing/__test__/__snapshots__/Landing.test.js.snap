// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> matches snapshot 1`] = `
Array [
  <div>
    <div>
      <h1>
        Encontre a propriedade ideal para você!
      </h1>
      <div
        className="Filters__container"
      >
        <div>
          <button
            onClick={[Function]}
            type="button"
          >
            <p
              label="propertyKind"
            >
              Tipo de Propriedade
            </p>
            <div>
              <p>
                Selecione...
              </p>
              <svg
                fill="currentColor"
                height={22}
                size={22}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width={22}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </button>
        </div>
        <div>
          <button
            onClick={[Function]}
            type="button"
          >
            <p>
              Área do Imóvel
            </p>
            <div>
              0 até 1
            </div>
          </button>
        </div>
        <div>
          <button
            onClick={[Function]}
            type="button"
          >
            <p>
              Estado
            </p>
            <div>
              <p>
                Selecione...
              </p>
              <svg
                fill="currentColor"
                height={22}
                size={22}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width={22}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </button>
        </div>
        <div>
          <button
            onClick={[Function]}
            type="button"
          >
            <p>
              Intervalo de Preço
            </p>
            <div>
              0 até 1
            </div>
          </button>
        </div>
        <button
          className="button searchButton"
          onClick={[Function]}
          type="button"
        >
          Procurar Imóvel
        </button>
      </div>
    </div>
  </div>,
  <div>
    <h2>
      O que você precisa?
    </h2>
    <div>
      <div>
        <div>
          <img
            alt="test"
            src="test-file-stub"
          />
          <div>
            <p>
              Anunciar Imóvel Rural
            </p>
            <p>
              Quero criar um anúncio para vender minha fazenda, chácara ou sítio.
            </p>
          </div>
        </div>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          <p>
            Anunciar
          </p>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </button>
      </div>
      <div>
        <div>
          <img
            alt="test"
            src="test-file-stub"
          />
          <div>
            <p>
              Comprar Imóvel Rural
            </p>
            <p>
              Tenho interesse em comprar um Imóvel rural.
            </p>
          </div>
        </div>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          <p>
            Comprar
          </p>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </button>
      </div>
      <div>
        <div>
          <img
            alt="test"
            src="test-file-stub"
          />
          <div>
            <p>
              Anúncio Profissional
            </p>
            <p>
              Quero contratar o serviço de criação de aunúncio profissional.
            </p>
          </div>
        </div>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          <p>
            Virar Pro
          </p>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  <div>
    <div />
    <h1>
      Funcionalidades e Benefícios do Terrafinder
    </h1>
    <div>
      <div>
        <h2>
          Conheça e aprenda a utilizar o Terrafinder
        </h2>
        <p>
          Você que quer vender seu imóvel rural, encontrou o lugar ideal para anuncia-lo. É simples, rápido e eficiente.
        </p>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Nesciunt deserunt quisquam accusantium accusamus aliquid laborum nam quasi earum quaerat voluptate ratione cumque tempora minima, nihil sequi. Culpa, ut ad? At?
        </p>
        <div>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            <p>
              Quero anunciar meu imóvel
            </p>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            <p>
              Quero achar uma propriedade
            </p>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div>
        <div>
          <div>
            <p>
              Pague apenas pelo tempo do seu anúncio
            </p>
            <p>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Ea eius eum praesentium quos delectus vero commodi ipsum molestiae quod vel nostrum rerum alias minima explicabo dignissimos, iste tempora nulla perferendis.
            </p>
          </div>
        </div>
        <div>
          <div>
            <p>
              Pague apenas pelo tempo do seu anúncio
            </p>
            <p>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Ea eius eum praesentium quos delectus vero commodi ipsum molestiae quod vel nostrum rerum alias minima explicabo dignissimos, iste tempora nulla perferendis.
            </p>
          </div>
        </div>
        <div>
          <div>
            <p>
              Pague apenas pelo tempo do seu anúncio
            </p>
            <p>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Ea eius eum praesentium quos delectus vero commodi ipsum molestiae quod vel nostrum rerum alias minima explicabo dignissimos, iste tempora nulla perferendis.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div>
    <h1>
      Conheça nosso planos de anúncio
    </h1>
    <h2>
      Escolha o que melhor se encaixa para você
    </h2>
    <div>
      <div
        style={
          Object {
            "zIndex": 0,
          }
        }
      >
        <div>
          <p
            className="text-lime-green"
          >
            Grátis
          </p>
          <p />
        </div>
        <div
          className="bg-lime-green"
        >
          <p>
            Básico
          </p>
          <ul>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              <span>
                Duração de 7 dias
              </span>
            </li>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              <span>
                Até 15 fotos
              </span>
            </li>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span>
                Sem demarcação do imóvel no mapa
              </span>
            </li>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span>
                Sem suporte
              </span>
            </li>
          </ul>
          <button
            className="text-lime-green"
            type="button"
          >
            Quero esse
          </button>
        </div>
      </div>
      <div
        style={
          Object {
            "zIndex": 500,
          }
        }
      >
        <div>
          <p
            className="text-yellow"
          >
            R$ 50
          </p>
          <p>
            por anúncio
          </p>
        </div>
        <div
          className="bg-yellow"
        >
          <p>
            PREMIUM
          </p>
          <ul>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              <span>
                Duração do anúncio de 90 dias
              </span>
            </li>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              <span>
                Visibilidade aumentada na plataforma
              </span>
            </li>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              <span>
                Até 30 fotos e 1 vídeo
              </span>
            </li>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span>
                Sem filmagens por drone nem tratamento de fotos
              </span>
            </li>
          </ul>
          <button
            className="text-yellow"
            type="button"
          >
            Quero esse
          </button>
        </div>
      </div>
      <div
        style={
          Object {
            "zIndex": 0,
          }
        }
      >
        <div>
          <p
            className="text-pink"
          >
            A definir
          </p>
          <p>
            Faça uma cotação com nossos especialistas
          </p>
        </div>
        <div
          className="bg-blue"
        >
          <p>
            Profissional
          </p>
          <ul>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              <span>
                Duração de até 365 dias
              </span>
            </li>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              <span>
                Maior visibilidade da plataforma
              </span>
            </li>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span>
                Gravação da propriedade com drones e tratamento de imagens
              </span>
            </li>
            <li>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span>
                Suporte exclusivo
              </span>
            </li>
          </ul>
          <button
            className="text-pink"
            type="button"
          >
            Quero esse
          </button>
        </div>
      </div>
    </div>
    <div>
      <p>
        <b>
          Gratuito:
        </b>
         Recomendado para iniciantes, é uma ótima forma de testar a plataforma. Você pode testar a plataforma. Você pode criar um anúncio gratuíto da sua propriedade, com as informações gerais do imóvel, e dura 7 dias.
      </p>
      <p>
        <b>
          Premium:
        </b>
         Recomendado para quem já tem conteúdos da propriedade como fotos, detalhes do imóvel como tipo de solo, atividades, topografia. Ele custa R$ 50,00 e seu anúncio ganha mais visibilidade dentro da plataforma.
      </p>
      <p>
        <b>
          Proffissional:
        </b>
         Entre em contato conosco para conversarmos e conhecermos melhor o seu imóvel rural. Iremos até a propriedade, colheremos as informações mais importanjtes dela e faremos excelentes gravações explorando o imóvel através de dones de ponta. Essa é a maneira mais fácil de vender.
      </p>
    </div>
  </div>,
]
`;
